pkg/services/ngalert/schedule/registry.go

```diff
@@ -336,6 +336,11 @@ func (r ruleWithFolder) Fingerprint() fingerprint {
 		writeInt(0)
 	}
 
+	if len(rule.NotificationSettings) > 0 {
+		binary.LittleEndian.PutUint64(tmp, uint64(rule.NotificationSettings[0].Fingerprint()))
+		writeBytes(tmp)
+	}
```

Comment:
let's use for loop and iterate through all settings. I think we should honor the interface everywhere but in compat layer that explicitly handles only the first element.

Reply:
Left a comment elsewhere but from the comment on the field this is a slice purely to trick `xorm` into rendering it to json, I don't think we intend on having >1 setting.